File: AndroidBasics/calculator/src/main/java/pan/alexander/calculator/di/RepositoryModule.java
package pan.alexander.calculator.di;

import dagger.Module;
import dagger.Provides;
import pan.alexander.calculator.data.HistoryRepositoryImplementation;
import pan.alexander.calculator.data.SettingsRepositoryImplementation;
import pan.alexander.calculator.domain.HistoryRepository;
import pan.alexander.calculator.domain.SettingsRepository;

@Module
public class RepositoryModule {
    @Provides
    public HistoryRepository providesHistoryRepository() {
        return new HistoryRepositoryImplementation();
    }

    @Provides
    public SettingsRepository providesSettingsRepository() {
        return new SettingsRepositoryImplementation();
    }
}

File: simlpe-weather/app/src/main/java/pan/alexander/simpleweather/di/RepositoryModule.kt
package pan.alexander.simpleweather.di

import dagger.Module
import dagger.Provides
import pan.alexander.simpleweather.data.CurrentWeatherRepositoryImpl
import pan.alexander.simpleweather.domain.CurrentWeatherRepository

@Module
class RepositoryModule {
    @Provides
    fun providesCurrentWeatherRepository(): CurrentWeatherRepository {
        return CurrentWeatherRepositoryImpl()
    }
}
